### ISSUE TITLE
Trailing slash in libdir, conflicting libs and generate_gir

### ISSUE THREAD

**Issue by heftig** [mesonbuild/meson#8593]
Downstream bug: https://gitlab.gnome.org/GNOME/gnome-characters/-/issues/86

libhandy: [meson.build](https://gitlab.gnome.org/GNOME/libhandy/-/blob/7b38a860ffcec6c2ad28153358cc3d037ddb618f/meson.build)
gnome-characters: [lib/meson.build](https://gitlab.gnome.org/GNOME/gnome-characters/-/blob/8fde0572d3eaa19166ec139ef4f2c1d168404e95/lib/meson.build)

- libhandy configures a trailing slash into its `libdir`, which ends up in the .pc file:
   ```cfg
   prefix=/usr
   libdir=${prefix}/lib
   includedir=${prefix}/include

   Name: Handy
   Description: Handy Mobile widgets
   Version: 1.2.0
   Requires: gtk+-3.0
   Libs: -L${prefix}/lib/ -lhandy-1
   Cflags: -I${includedir}/libhandy-1
   ```
 - There's an unrelated library `/usr/lib/libgc.so` (Boehm GC).
 - gnome-characters builds a private library named `libgc.so` and introspection for it.

This situation causes meson to emit an additional `-L/usr/lib/` for g-ir-scanner, which is then translated into `-L/usr/lib/ -Wl,-rpath,/usr/lib/` for gcc and causes the link to fail as it prefers `/usr/lib/libgc.so` over the local `./libgc.so`.

Where is the bug here?

 - In libhandy for configuring the libdir to `/usr/lib/`?
 - In meson for not stripping the trailing slash when building libhandy?
 - In meson for not recognizing `/usr/lib/` as a duplicate of `/usr/lib` when building gnome-characters?
 - In gnome-characters for attempting to build a library named `gc`?
 - In GIR?

**Log:**
```
+ exec meson setup --prefix /usr --libexecdir lib --sbindir bin --buildtype plain --auto-features enabled --wrap-mode nodownload -D b_lto=true -D b_pie=true gnome-characters build
The Meson build system
Version: 0.57.1
Source dir: /build/gnome-characters/src/gnome-characters
Build dir: /build/gnome-characters/src/build
Build type: native build
Project name: gnome-characters
Project version: 40.beta
C compiler for the host machine: cc (gcc 10.2.0 "cc (GCC) 10.2.0")
C linker for the host machine: cc ld.bfd 2.36.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Checking if "libunistring support" links: YES 
Found pkg-config: /usr/bin/pkg-config (1.7.3)
Run-time dependency gjs-1.0 found: YES 1.67.2
Run-time dependency libhandy-1 found: YES 1.1.90
Run-time dependency gio-2.0 found: YES 2.67.4
Run-time dependency gtk+-3.0 found: YES 3.24.26
Run-time dependency pango found: YES 1.48.2
Run-time dependency pangoft2 found: YES 1.48.2
Found pkg-config: /usr/bin/pkg-config (1.7.3)
Program glib-compile-resources found: YES (/usr/bin/glib-compile-resources)
Program glib-compile-resources found: YES (/usr/bin/glib-compile-resources)
Configuring org.gnome.Characters.desktop.in using configuration
WARNING: Project targeting '>= 0.46.0' but tried to use feature introduced in '0.50.0': install arg in configure_file.
Configuring org.gnome.Characters.service using configuration
Configuring org.gnome.Characters.BackgroundService.service using configuration
Configuring org.gnome.Characters.search-provider.ini using configuration
Program glib-mkenums found: YES (/usr/bin/glib-mkenums)
Build-time dependency gobject-introspection-1.0 found: YES 1.66.1
Dependency gobject-introspection-1.0 found: YES 1.66.1 (cached)
Program g-ir-scanner found: YES (/usr/bin/g-ir-scanner)
Dependency gobject-introspection-1.0 found: YES 1.66.1 (cached)
Program g-ir-compiler found: YES (/usr/bin/g-ir-compiler)
Program gjs found: YES (/usr/bin/gjs)
Configuring org.gnome.Characters using configuration
Configuring org.gnome.Characters.BackgroundService using configuration
Program glib-compile-resources found: YES (/usr/bin/glib-compile-resources)
Program glib-compile-resources found: YES (/usr/bin/glib-compile-resources)
Configuring config.h using configuration
Program meson_post_install.py found: YES (/usr/bin/python /build/gnome-characters/src/gnome-characters/meson_post_install.py)
Build targets in project: 14
WARNING: Project specifies a minimum meson_version '>= 0.46.0' but uses features which were added in newer versions:
 * 0.50.0: {'install arg in configure_file'}

Option buildtype is: plain [default: debugoptimized]
Found ninja-1.10.2 at /usr/bin/ninja
ninja: Entering directory `build'
[1/13] Generating org.gnome.Characters.src_gresource with a custom command
[2/13] Generating org.gnome.Characters.BackgroundService.data_gresource with a custom command
[3/13] Generating org.gnome.Characters.BackgroundService.src_gresource with a custom command
[4/13] Generating org.gnome.Characters.data_gresource with a custom command
[5/13] Generating org.gnome.Characters.appdata.xml_data_merge with a custom command
[6/13] Generating org.gnome.Characters.desktop_data_merge with a custom command
[7/13] Generating gc-enumtypes.h with a custom command (wrapped by meson to capture output)
[8/13] Generating gc-enumtypes.c with a custom command (wrapped by meson to capture output)
[9/13] Compiling C object lib/libgc.so.p/gc.c.o
../gnome-characters/lib/gc.c: In function ‘gc_search_criteria_copy’:
../gnome-characters/lib/gc.c:624:3: warning: ‘g_memdup’ is deprecated: Use 'g_memdup2' instead [-Wdeprecated-declarations]
  624 |   GcSearchCriteria *criteria = g_memdup (boxed, sizeof (GcSearchCriteria));
      |   ^~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:82,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from ../gnome-characters/lib/gc.h:7,
                 from ../gnome-characters/lib/gc.c:3:
/usr/include/glib-2.0/glib/gstrfuncs.h:257:23: note: declared here
  257 | gpointer              g_memdup         (gconstpointer mem,
      |                       ^~~~~~~~
[10/13] Compiling C object lib/libgc.so.p/meson-generated_.._gc-enumtypes.c.o
[11/13] Linking target lib/libgc.so
[12/13] Generating Gc-1.0.gir with a custom command
FAILED: lib/Gc-1.0.gir 
/usr/bin/g-ir-scanner --no-libtool --namespace=Gc --nsversion=1.0 --warn-all --output lib/Gc-1.0.gir --c-include=gc.h -I/build/gnome-characters/src/gnome-characters/lib -I/build/gnome-characters/src/build/lib -I./. -I../gnome-characters/. --filelist=/build/gnome-characters/src/build/lib/libgc.so.p/Gc_1.0_gir_filelist --include=Gio-2.0 --include=Gtk-3.0 --symbol-prefix=gc --identifier-prefix=Gc --cflags-begin -I./. -I../gnome-characters/. -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/lzo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cloudproviders -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/at-spi-2.0 -I/usr/include/libhandy-1 -I/usr/include/gobject-introspection-1.0 -D_FORTIFY_SOURCE=2 --cflags-end --add-include-path=/usr/share/gir-1.0 -L./lib --library gc --extra-library=gio-2.0 --extra-library=gobject-2.0 --extra-library=glib-2.0 --extra-library=gtk-3 --extra-library=gdk-3 --extra-library=z --extra-library=pangocairo-1.0 --extra-library=pango-1.0 --extra-library=harfbuzz --extra-library=atk-1.0 --extra-library=cairo-gobject --extra-library=cairo --extra-library=gdk_pixbuf-2.0 -L/usr/lib/ --extra-library=handy-1 --extra-library=pangoft2-1.0 --extra-library=fontconfig --extra-library=freetype --extra-library=girepository-1.0 --sources-top-dirs /build/gnome-characters/src/gnome-characters/subprojects/ --sources-top-dirs /build/gnome-characters/src/build/subprojects/
In file included from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/limits.h:26,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include-fixed/limits.h:195,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include-fixed/syslimits.h:7,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include-fixed/limits.h:34,
                 from /usr/lib/glib-2.0/include/glibconfig.h:11,
                 from /usr/include/glib-2.0/glib/gtypes.h:32,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /build/gnome-characters/src/gnome-characters/lib/gc.h:7,
                 from /build/gnome-characters/src/build/g-ir-cpp-0zp9x6av.c:4:
/usr/include/features.h:397:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
  397 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~
g-ir-scanner: link: gcc -pthread -o /build/gnome-characters/src/build/tmp-introspectigy4o1mb/Gc-1.0 -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/build/gnome-characters/src=/usr/src/debug /build/gnome-characters/src/build/tmp-introspectigy4o1mb/Gc-1.0.o -L. -Wl,-rpath,. -Wl,--no-as-needed -L./lib -L/usr/lib/ -Wl,-rpath,/usr/lib/ -lgc -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lgtk-3 -lgdk-3 -lz -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lhandy-1 -lpangoft2-1.0 -lfontconfig -lfreetype -lgirepository-1.0 -lgio-2.0 -lgobject-2.0 -Wl,--export-dynamic -lgmodule-2.0 -pthread -lglib-2.0 -lglib-2.0 -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
/build/gnome-characters/src/build/tmp-introspectigy4o1mb/Gc-1.0: symbol lookup error: /build/gnome-characters/src/build/tmp-introspectigy4o1mb/Gc-1.0: undefined symbol: gc_search_context_get_type
Command '['/build/gnome-characters/src/build/tmp-introspectigy4o1mb/Gc-1.0', '--introspect-dump=/build/gnome-characters/src/build/tmp-introspectigy4o1mb/functions.txt,/build/gnome-characters/src/build/tmp-introspectigy4o1mb/dump.xml']' returned non-zero exit status 127.
ninja: build stopped: subcommand failed.
```
